Entity/BaseProduct.php

```diff
@@ -1045,18 +1045,43 @@ public function setSearchTerms($searchTerms)
     /**
      * Helper method to check if the product is
      * a valid shop product.
-     *
+     * @param $defaultCurrency
      * @return bool
      */
-    public function isValidShopProduct()
+    public function isValidShopProduct($defaultCurrency)
     {
         if (method_exists($this, 'getPrices') &&
             $this->getStatus()->getId() == Status::ACTIVE &&
             $this->getPrices() && count($this->getPrices()) > 0
         ) {
+            if(!$this->hasPriceInDefaultCurrency($this->getPrices(), $defaultCurrency)) {
```

Comment:
+            if($this->hasPriceInDefaultCurrency($this->getPrices(), $defaultCurrency)) {
 +                $isValid = true;
 +            }
 +            if($this->getSupplier()) {
 +                $isValid = true;
 +            }
 +          }
  
          return $isValid;
      }